components/MermaidDiagram.tsx

```
@@ -0,0 +1,119 @@
+    return (
+        <div
+            className={cn(styles.mermaidWrapper, className)}
+            role="img"
+            aria-label={ariaLabel}
+            aria-roledescription="diagram"
+        >
+            <div ref={containerRef} className={styles.mermaidTarget} aria-hidden={rendered ? undefined : 'true'} />
+            {!rendered && !error && (
+                <pre className="codeblock" aria-hidden="true">
+                    {chart.split('\n').map((line, idx) => (
+                        <div className="code-line" key={`${idx}-${line}`}>
```

Biome (2.4.15)

[error] Avoid using the index of an array as key property in an element.

(lint/suspicious/noArrayIndexKey)